src/conways.rs

```diff
@@ -6,42 +6,62 @@ enum State {
 
 type Position = (usize, usize);
 
-struct Grid {
+pub struct Grid {
     cells: Vec<Vec<State>>,
 }
 
+
 impl Grid {
     pub fn new(size: usize) -> Grid {
         Grid {
             cells: vec![vec![State::Dead; size]; size],
         }
     }
 
-    pub fn change_state(&mut self, position: Position, state: State) {
+    fn change_state(&mut self, position: Position, state: State) {
         let (x, y) = position;
 
-        let outer = &mut self.cells[y];
-        let inner = &mut outer[x];
+        let outer = &mut self.cells[x];
+        let inner = &mut outer[y];
 
         *inner = state;
     }
 
-    fn get_cell(&self, position: Position) -> State {
+    fn get_state(&self, position: Position) -> State {
         let (x, y) = position;
 
-        let outer = &self.cells[y];
-        let state = outer[x];
+        let outer = &self.cells[x];
+        let state = outer[y];
         return state;
     }
 
-    // pub fn update(&mut self) {
-    // }
+    fn determine_new_state(&self, position: Position, current_state: &State) -> State {
+        let alive_neighbours = self.alive_neighbors_amount(position);
+
+        if *current_state == State::Alive {
+            if alive_neighbours < 2 {
+                State::Dead
+            } else if alive_neighbours == 2 || alive_neighbours == 3 {
+                State::Alive
+            } else if alive_neighbours > 3 {
+                State::Dead
+            } else {
+                State::Dead
+            }
+        } else {
+            if alive_neighbours == 3 {
+                State::Alive
+            } else {
+                State::Dead
+            }
+        }
```

Comment:
What do you think if instead of using the if-else you use a match?

Something like this:

```Rust
match (current_state, alive_neighbours) {
     (State::Alive, 2 | 3) => State::Alive, 
     ...
}
```

Reply:
Wow! That's a very neat feature. Thank you very much ^_^.

Thank you very _match_